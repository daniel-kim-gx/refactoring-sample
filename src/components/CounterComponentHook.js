import React, { useState } from "react";
import Bolder from "../base/Bolder";

const CounterComponentHook = () => {
  const [count, setCount] = useState(0);

  const increase = () => setCount((count) => count + 1);

  return (
    <div>
      <div>
        <Bolder>counter : {count}</Bolder>
      </div>
      <button onClick={increase}>Increase</button>
    </div>
  );
};

export default CounterComponentHook;
