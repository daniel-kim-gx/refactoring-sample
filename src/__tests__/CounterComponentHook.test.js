import { render } from "@testing-library/react";
import CounterComponentHook from "../components/CounterComponentHook";

describe("<CounterComponentHook />", () => {
  it("Should match snapshot", () => {
    const { container } = render(<CounterComponentHook />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
