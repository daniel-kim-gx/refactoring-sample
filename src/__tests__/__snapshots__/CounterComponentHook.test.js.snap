// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterComponentHook /> Should match snapshot 1`] = `
<div>
  <div>
    <b>
      counter : 
      0
    </b>
  </div>
  <button>
    Increase
  </button>
</div>
`;
